tests/test_footprintMergeCatalog.py

```diff
@@ -102,35 +102,49 @@ def setUp(self):
         insertPsf(pos3, im3, psf3, kernelSize, flux)
 
         schema = afwTable.SourceTable.makeMinimalSchema()
-        self.idFactory = afwTable.IdFactory.makeSimple()
-        self.table = afwTable.SourceTable.make(schema, self.idFactory)
 
         # Create SourceCatalogs from these objects
         fp1 = afwDetect.FootprintSet(
             im1, afwDetect.Threshold(0.001), "DETECTED")
-        self.catalog1 = afwTable.SourceCatalog(self.table)
+        # Use a different ID factory for each footprint so that the peak ID's
+        # overlap. This allows us to test ID collisions.
+        idFactory = afwTable.IdFactory.makeSimple()
+        table = afwTable.SourceTable.make(schema, idFactory)
+        self.catalog1 = afwTable.SourceCatalog(table)
         fp1.makeSources(self.catalog1)
 
+        idFactory = afwTable.IdFactory.makeSimple()
+        table = afwTable.SourceTable.make(schema, idFactory)
         fp2 = afwDetect.FootprintSet(
             im2, afwDetect.Threshold(0.001), "DETECTED")
-        self.catalog2 = afwTable.SourceCatalog(self.table)
+        self.catalog2 = afwTable.SourceCatalog(table)
         fp2.makeSources(self.catalog2)
 
+        idFactory = afwTable.IdFactory.makeSimple()
+        table = afwTable.SourceTable.make(schema, idFactory)
         fp3 = afwDetect.FootprintSet(
             im3, afwDetect.Threshold(0.001), "DETECTED")
-        self.catalog3 = afwTable.SourceCatalog(self.table)
+        self.catalog3 = afwTable.SourceCatalog(table)
         fp3.makeSources(self.catalog3)
 
     def tearDown(self):
         del self.catalog1
         del self.catalog2
         del self.catalog3
-        del self.table
+
+    def assertUniqueIds(self, catalog):
+        """Ensure that all of the peak IDs in a single parent are unique
+        """
+        for src in catalog:
+            # The peak catalog may be non-contiguous,
+            # so add each peak individually
+            peaks = [peak.getId() for peak in src.getFootprint().peaks]
+            self.assertEqual(len(peaks), len(np.unique(peaks)))
```

Comment:
How about `self.assertEqual(sorted(peaks), sorted(set(peaks))`? That would specify exactly which ones don't match.